philipmeneghini/InstaConnect
# Lazy Loading For Home Page

**Summary**
Currently the home page load is extremely slow due to the way it is calling the api and iterating through all users following and all their associated contents. In this PR I created more effective endpoints to get all users that a person is following and then lazy load the contents from the backend from these users. Since we are calling these endpoints once in the `useEffect` hook there is a dramatic speed improvement. As we scroll we can see new backend requests are made to retrieve more posts that users following may have made.

![HomePageLazyLoad](https://github.com/philipmeneghini/InstaConnect/assets/63440066/c5c3d523-0020-4e83-84ea-cdc998c7b981)

This will be the first of a series of PRs that will add more lazy loading throughout the application. Next I will lazily load comments and likes on each post.

**Testing**

- Make sure that new content is called as scrolling down the homepage
- content should be ordered by date created
- verify new endpoints work for users and contents. Try passing an index and limit parameter to only call in a small subsect of the data
